esmvaltool/cmorizers/obs/cmorize_obs_woa.py

```diff
@@ -40,31 +41,84 @@ def _fix_data(cube, var):
-                cube, var, out_dir, attrs, unlimited_dimensions=['time'])
+def collect_files(in_dir, var, cfg):
+    """Compose input file list and download if missing."""
+    file_list = []
+    var_dict = cfg['variables'][var]
+    in_dir = os.path.join(in_dir, var_dict['name'])
+    # create list of monthly data
+    for mon in range(1, 13):
+        fname = var_dict['file'] + "{:02d}".format(mon) + '_01.nc'
+        in_file = os.path.join(in_dir, fname)
+
+        # download if missing
+        if not os.path.isfile(in_file):
+            if not os.path.isdir(in_dir):
+                os.makedirs(in_dir)
+            logger.info('Input file %s is missing. Start download ... ', fname)
+            url = os.path.join(cfg['attributes']['source'], var_dict['name'],
+                               'netcdf', var_dict['file'].split('_')[1],
+                               cfg['custom']['resolution'], fname)
+            url_file = requests.get(url)
+            open(in_file, 'wb').write(url_file.content)
+
+        file_list.append(in_file)
+
+    return file_list
+
+
+def extract_variable(in_files, out_dir, attrs, raw_info, cmor_table):
+    """Extract variables and create OBS dataset."""
+    var = raw_info['var']
+    var_info = cmor_table.get_variable(raw_info['mip'], var)
+    rawvar = raw_info['raw_var']
+    with catch_warnings():
+        filterwarnings(
+            action='ignore',
+            message='Ignoring netCDF variable .* invalid units .*',
+            category=UserWarning,
+            module='iris',
+        )
+        cubes = iris.load(in_files, rawvar)
+    iris.util.equalise_attributes(cubes)
+    cube = cubes.concatenate_cube()
+
+    # set reference time
+    year = raw_info['reference_year']
+    cube.coord('time').points = np.arange(0.5, 12.5)
+    cube.coord('time').units = Unit('months since ' + str(year) +
+                                    '-01-01 00:00:00',
+                                    calendar='gregorian')
+
+    fix_var_metadata(cube, var_info)
+    fix_coords(cube)
+    _fix_data(cube, var)
+    set_global_atts(cube, attrs)
+    save_variable(cube, var, out_dir, attrs, unlimited_dimensions=['time'])
+
+    # compute derived vars
+    if 'srf_var' in raw_info:
+        var_info = cmor_table.get_variable(raw_info['mip'],
+                                           raw_info['srf_var'])
+        logger.info("Extract surface OBS for %s", raw_info['srf_var'])
+        level_constraint = iris.Constraint(cube.var_name, depth=0)
+        cube = cube.extract(level_constraint)
+        fix_var_metadata(cube, var_info)
+        save_variable(cube,
+                      raw_info['srf_var'],
+                      out_dir,
+                      attrs,
+                      unlimited_dimensions=['time'])
+
+    if raw_info['compute_Oyr']:
+        attrs['mip'] = 'Oyr'
+        cube = annual_statistics(cube, 'mean')
+        save_variable(cube, var, out_dir, attrs, unlimited_dimensions=['time'])
```

Comment:
With the current implementation, yearly data are only computed for the surface level. The same cube is used to extract the surface layer and compute the yearly mean. These would need to be separated if the plan is to compute Omon, Oyr and Omon surface data in one go.

Reply:
Actually you are right, here i mixed up a bit different things... but we have cmor tables as our main guideline! 
`Omon`: thetao, so, no3, o2, po4, si, tos, sos, no3os, o2os, po4os, sios
`Oyr`: thetao, so, no3, o2, po4, si (no surface variables in this table)
First guess ... we need to create an intermediate cube when computing surface variables and preserve the original one for the following computation of annual values.

Here I made the annual average an option (`compute_Oyr`), but probably the option should be removed and annual mean data always computed.

Reply:
I aggressed this issue in 84dfb52 and now also documentation should go along well...